tmt/steps/provision/__init__.py

```diff
@@ -449,32 +449,39 @@ def _ansible_playbook_path(self, playbook):
         self.debug(f"Playbook full path: '{playbook}'", level=2)
         return playbook
 
-    def _export_environment(self, execute_environment=None):
-        """ Prepare shell export of environment variables """
+    def _prepare_environment(self, execute_environment=None):
+        """ Prepare dict of environment variables """
         # Prepare environment variables so they can be correctly passed
         # to ssh's shell. Create a copy to prevent modifying source.
         environment = dict()
         environment.update(execute_environment or dict())
         # Plan environment and variables provided on the command line
         # override environment provided to execute().
         environment.update(self.parent.plan.environment)
-        # Prepend with export and run as a separate command.
-        if not environment:
-            return ''
-        return 'export {}; '.format(
-            ' '.join(tmt.utils.shell_variables(environment)))
+        return {} if not environment else environment
+
+    @staticmethod
+    def _export_environment(environment):
+        """ Prepare shell export of environment variables """
+        return f'export {" ".join(tmt.utils.shell_variables(environment))}; ' \
+            if environment \
+            else ''
 
     def ansible(self, playbook, extra_args=None):
         """ Prepare guest using ansible playbook """
         playbook = self._ansible_playbook_path(playbook)
+        self.run(f'stty cols {tmt.utils.OUTPUT_WIDTH};:',
+                 shell=True)
+        ssh_common_args = [f'--ssh-common-args="{i}"'
+                           for i in self._ssh_options(join=True).split()]
         stdout, stderr = self.run(
-            f'{self._export_environment()}'
-            f'stty cols {tmt.utils.OUTPUT_WIDTH}; ansible-playbook '
-            f'--ssh-common-args="{self._ssh_options(join=True)}" '
+            f'ansible-playbook '
             f'{self._ansible_verbosity()} '
-            f'{self._ansible_extra_args(extra_args)} -i {self._ssh_guest()},'
-            f' {playbook}',
-            cwd=self.parent.plan.worktree)
+            f'{self._ansible_extra_args(extra_args)}'.split() +
```

Comment:
Shouldn't `shlex.split` be used instead? Otherwise user has no way how to escape space if necessary.